src/main/java/io/quarkus/search/app/quarkusio/QuarkusIO.java

```diff
@@ -95,6 +100,39 @@ public QuarkusIO(QuarkusIOConfig config, GitCloneDirectory mainRepository,
         this.localizedSiteUris = localizedSites.entrySet().stream().collect(
                 Collectors.toUnmodifiableMap(Map.Entry::getKey, e -> config.localized().get(e.getKey().code).webUri()));
         this.failureCollector = failureCollector;
+
+        this.localizedSitesUpToDate = Collections.unmodifiableMap(
+                createUpToDateLocalizedSitesMap(mainRepository, localizedSites, failureCollector));
+    }
+
+    private static Map<Language, Boolean> createUpToDateLocalizedSitesMap(
+            GitCloneDirectory mainRepository,
+            Map<Language, GitCloneDirectory> localizedSites,
+            FailureCollector failureCollector) {
+        Map<Language, Boolean> localizedSitesUpToDate = new HashMap<>();
+        ObjectId latestMainRev = mainRepository.currentRev();
+        for (Map.Entry<Language, GitCloneDirectory> localized : localizedSites.entrySet()) {
+            localized.getValue().currentUpstreamRev().ifPresent(revId -> {
```

Comment:
And if absent... ? Looks like you don't handle the null case when you read the map, so this would fail hard.